opsimulator/opsimulator.go

```diff
@@ -510,7 +515,22 @@ func (opSim *OpSimulator) checkInteropInvariants(ctx context.Context, logs []typ
 	return nil
 }
 
-// Overridden such that the port field can appropiately be set
+// validateDependencySet checks if the source chain is allowed to send messages to the executing chain
+func (opSim *OpSimulator) validateDependencySet(sourceChainID uint64) error {
+	executingChainID := opSim.Config().ChainID
+	executingChainDependencySet := opSim.Config().L2Config.DependencySet
+
+	// Check if source chain is in executing chain's dependency set
+	for _, depChainID := range executingChainDependencySet {
+		if depChainID == sourceChainID {
+			return nil // Source chain is in dependency set
+		}
+	}
+
+	return fmt.Errorf("executing message in block (chain %d) may not execute message from chain %d: not in dependency set", executingChainID, sourceChainID)
```

Comment:
🤔 Does throwing this error create a false sense security, or is it beneficial to dev testing?

Reply:
I think this is good. it is helpful because in prod the tx will be dropped, so end behavior is still the same, this just adds additional logging.